materialize-databricks/driver.go

```diff
@@ -0,0 +1,682 @@
+package main
+
+import (
+	"os"
+  "sync"
+	"context"
+	stdsql "database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"strings"
+	"text/template"
+	"time"
+
+	_ "github.com/databricks/databricks-sql-go"
+	dbsqlerr "github.com/databricks/databricks-sql-go/errors"
+	"github.com/databricks/databricks-sdk-go"
+	dbConfig "github.com/databricks/databricks-sdk-go/config"
+	driverctx "github.com/databricks/databricks-sql-go/driverctx"
+	boilerplate "github.com/estuary/connectors/materialize-boilerplate"
+	sql "github.com/estuary/connectors/materialize-sql"
+	pf "github.com/estuary/flow/go/protocols/flow"
+	pm "github.com/estuary/flow/go/protocols/materialize"
+	log "github.com/sirupsen/logrus"
+	"go.gazette.dev/core/consumer/protocol"
+	"golang.org/x/sync/errgroup"
+)
+
+const defaultPort = "443"
+const volumeName = "flow_staging"
+
+type tableConfig struct {
+	Table         string `json:"table" jsonschema:"title=Table,description=Name of the table" jsonschema_extras:"x-collection-name=true"`
+	Schema        string `json:"schema" jsonschema:"title=Schema,description=Schema where the table resides,default=default"`
```

Comment:
For consistency with other materializations and usability, I suggest:
- Not setting a default for `schema` in `tableConfig`
- Making `schema` optional in `tableConfig`
- In `config`, make `schema_name` a required field (note: I am assuming that you can't connect without some kind of schema specified`, and maybe make the name of the equivalent fields in the `tableConfig` and `config` be the same
- In `(*config).Validate`, error if `schema_name` hasn't been provided
- In `newTableConfig`, by default initialize the schema to the schema from `config`, which will be over-written if there is a present `schema` property in the raw json of the resource config when it is unmarshalled elsewhere

This would support the case where most bindings use the same schema as set in the endpoint-level `config`, but just a few need to be edited to use a different schema, which can be done by just configuring it for those specific bindings. It would directly parallel the BigQuery and Snowflake materializations, which work in this way. As it is now, if you wanted to materialize all the bindings to a schema other than "default", you would need to set it for every single binding.